bson/src/main/org/bson/codecs/configuration/mapper/ClassModel.java

```diff
@@ -34,6 +35,10 @@
     private final Map<String, FieldModel> fields = new TreeMap<String, FieldModel>();
     private final CodecRegistry registry;
     private final TypeResolver resolver;
+    private final MemberResolver memberResolver;
+    private final WeightedValue<String> collectionName = new WeightedValue<String>();
```

Comment:
So, perhaps the solution here isn't to add something explicit into the code, but rather to add a way for ODMs to annotate things at runtime with desirable information. For instance, have a Map<string, string> or something.